File: Programmers/131128.java
import java.util.*;

class Solution {
	
	static int[] cntx = new int[10];
	static int[] cnty = new int[10];
    
    public String solution(String x, String y) {
		StringBuilder answer = new StringBuilder();
		
		int count;
		for(int i=0; i<x.length(); i++) {
			++cntx[x.charAt(i)-'0'];
		}
		for(int i=0; i<y.length(); i++) {
			++cnty[y.charAt(i)-'0'];
		}
		for(int i=9; i>0; i--) {
			count = Math.min(cntx[i],cnty[i]);
			for(int j=0; j<count; j++) {
				answer.append(i);
			}
		}
		count = Math.min(cntx[0],cnty[0]);
		if(answer.length() == 0 && count>0) {
			answer.append(0);
		} else {
			for(int j=0; j<count; j++) {
				answer.append(0);
			}
		}
		
		if(answer.length() == 0) {
			answer.append(-1);
		}
        return answer.toString();
    }
}

File: Programmers/131128.cpp
#include <string>
#include <map>
#include <string>
#include <algorithm>
using namespace std;

map<int, int> mx;
map<int, int> my;
int cnt[10];
using namespace std;

string solution(string x, string y) {
    string s = "";
	for (auto q : x) {
		++mx[q - '0'];
	}
	for (auto q : y) {
		++my[q - '0'];
	}
	for (int i = 9; i > 0; i--) {
		int count = min(mx[i], my[i]);
		for (int j = 0; j < count; j++) {
			s += to_string(i);
		}
	}
	int count = min(mx[0], my[0]);
	if (s == "" && count > 0) {
		s = "0";
	} else {
		for (int j = 0; j < count; j++) {
			s += "0";
		}
	}
	if (s == "") {
		s = "-1";
	}
    return s;
}
